### PR TITLE
ignore stale qemu.pid files and don't invent backends

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

File: script/worker [hunk 1]
```diff
@@ -120,7 +120,6 @@ sub read_worker_config($){
             }
         }
     }
-    $sets->{'BACKEND'} ||= "qemu";
```
Comment (aaannz) [inline script/worker]: If someone does not provide BACKEND in workers.ini, I think this will trigger uninitialized variable during register_worker:338

**Comment by coolo** [comment]
the variable is just assigned, but not used? So 338 definitely does not give a warning - possibly later?

**Comment by coolo** [comment]
there are btw tons of warnings, but I don't think it's related

<pre>
Mär 21 03:07:10 ix64ph1072 worker[24134]: Use of uninitialized value $_[1] in string at /usr/lib/perl5/vendor_perl/5.18.2/Mojo/Util.pm line 111.
Mär 21 03:07:10 ix64ph1072 worker[24134]: Use of uninitialized value $_[1] in string at /usr/lib/perl5/vendor_perl/5.18.2/Mojo/Util.pm line 111.
Mär 21 03:07:10 ix64ph1072 worker[24134]: Use of uninitialized value $_[1] in string at /usr/lib/perl5/vendor_perl/5.18.2/Mojo/Util.pm line 111.
Mär 21 04:26:42 ix64ph1072 worker[24134]: Use of uninitialized value $_[1] in string at /usr/lib/perl5/vendor_perl/5.18.2/Mojo/Util.pm line 111.
Mär 21 04:26:42 ix64ph1072 worker[24134]: Use of uninitialized value $_[1] in string at /usr/lib/perl5/vendor_perl/5.18.2/Mojo/Util.pm line 111.
Mär 21 04:26:42 ix64ph1072 worker[24134]: Use of uninitialized value $_[1] in string at /usr/lib/perl5/vendor_perl/5.18.2/Mojo/Util.pm line 111.
Mär 21 04:26:48 ix64ph1072 worker[24134]: Use of uninitialized value $op in numeric eq (==) at /usr/lib/perl5/vendor_perl/5.18.2/Mojo/Transaction/WebSocket.pm line 325.
</pre>

**Comment by coolo** [comment]
the other worker only shows
<pre>
Mär 22 19:36:09 ix64ph1073 worker[10191]: Use of uninitialized value $op in numeric eq (==) at /usr/lib/perl5/vendor_perl/5.18.2/Mojo/Transaction/WebSocket.pm line 325.
</pre>

So possibly the warning *is* related

**Comment by coolo** [comment]
The warning is our least problem ;(

<pre>
[Mon Mar 23 07:25:08 2015] [error] DBIx::Class::Storage::DBI::_dbh_execute(): DBI Exception: DBD::Pg::st execute failed: FEHLER:  NULL-Wert in Spalte „backend“ verletzt Not-Null-Constraint
DETAIL:  Fehlgeschlagene Zeile enthält (23, linux.site, 1, null, 2015-03-23 06:25:08, 2015-03-23 06:25:08). [for Statement "INSERT INTO workers ( backend, host, instance, t_created, t_updated) VALUES ( ?, ?, ?, ?, ? ) RETURNING id" with ParamValues: 1=undef, 2='linux.site', 3='1', 4='2015-03-23 06:25:08', 5='2015-03-23 06:25:08'] at /home/coolo/prod/openQA/script/../lib/OpenQA/Scheduler.pm line 161
</pre>

**Comment by coolo** [comment]
our schema is broken - the backend can very well be NULL and is mainly a worker capability.

**Comment by lnussel** [comment]
I'm not sure the backend changes are a good idea. This needs more thinking about changing the architecture. The current architecture should work even though there may be some missing checks to make sure a qemu worker actually only gets qemu jobs.

**Comment by coolo** [comment]
I will split the PR into 3 - it's seperate topics actually

**Comment by coolo** [comment]
299-301

**Pull request closed without merging**